### PR TITLE
Add some model unit tests

### PR DESCRIPTION
**Pull request by brianhelba**

### CODE REVIEW

File: publish/tests/test_version.py [hunk 1]
```diff
@@ -1,8 +1,25 @@
 import pytest
 
+from publish.models import Version
 from .fuzzy import DandisetIdentifierRe, TimestampRe, VersionRe
 
 
+@pytest.mark.django_db
+def test_version_make_version_nosave(dandiset, version_factory):
```
Comment (dchiquit) [inline publish/tests/test_version.py:8]: If I had to guess, it looks like this is a factoryboy fixture. Is it required for this test, or does it inject something somewhere?

File: publish/tests/test_version.py [hunk 2]
```diff
@@ -1,8 +1,25 @@
...
+    # Without saving, the output should be reproducable
+    version_str_1 = Version.make_version(dandiset)
+    version_str_2 = Version.make_version(dandiset)
+    assert version_str_1 == version_str_2
+
+
+@pytest.mark.django_db
+def test_version_make_version_save(dandiset, version_factory):
+    # Given an existing version at the current time, a different one should be allocated
+    version_1 = version_factory(dandiset=dandiset)
```
Comment (dchiquit) [inline publish/tests/test_version.py:18]: Have you verified that `version_factory` will actually invoke `make_version` when initializing the new object? I can imagine this test having false negatives if it just guesses a valid-looking version string.

File: publish/tests/test_version.py (hunk 1 above)
Comment (brianhelba) [reply publish/tests/test_version.py:8]: You're right, `version_factory` is unnecessary in this case.

File: publish/tests/test_version.py (hunk 2 above)
Comment (brianhelba) [reply publish/tests/test_version.py:18]: It's a good point. I can add a Spy to ensure it's called.

File: publish/tests/test_version.py (hunk 1 above)
Comment (brianhelba) [reply publish/tests/test_version.py:8]: @dchiquit And yes, these fixtures are defined by https://pytest-factoryboy.readthedocs.io/ , which is small but very helpful.

**Comment by brianhelba** [comment]
@dchiquit Fixed.

**Pull request merged**